<!-- -->

<launch>
	
	<!-- uses sim time (rosbag time) -->
	<param name="/use_sim_time" value="True"/>

	<!-- FME Big Door environment urdf launch file -->
<!-- 	<param name="fme/robot_description" command="cat $(find tibi_dabo_description)/urdf/fme.urdf" /> -->
 	<include file="$(find tibi_dabo_base)/launch/fme.launch"/> -->

	<!-- Rosbag: Platform, Odometry, front and rear lasers and clock in FME Big Door environment-->
	<node	pkg="rosbag"
		type="rosbag"
		name="rosbag"
		args="play --clock /home/andreu/rosbags/2011-08-24-16-06-32.bag" />

	<!-- 	Launch localization node inside a namespace defining robot name -->
	<group ns="$(env ROBOT)">
		<!-- segway urdf model -->
		<param name="robot_description" command="cat $(find tibi_dabo_base)/model/tibi_dabo_base.urdf" />
		
		<!-- map-based localization node -->
		<node	pkg="iri_localization3d"
			type="iri_localization3d"
			name="iri_localization3d"
			args=""
			output="screen">

			<!--localization parameters -->
			<param name="num_particles" value= "50" type="int" />
			<param name="initX" value= "9.0" type="double" />  
			<param name="initY" value= "11.0" type="double" /> 
			<param name="initH" value= "0.0" type="double" /> 
			<param name="mapFile" value= "/home/andreu/iriSw/faramotics/models/fmeBigDoor.obj" type="str" />
			<param name="floorGridFile" value= "/home/andreu/iriSw/faramotics/models/fmeBigDoor_zGrid_20cm.txt" type="str" />  
			<param name="l1_frameName" value= "front_laser" type="str" />
			<param name="l1_nRays" value= "720" type="int" />
			<param name="l1_aperture" value= "3.14" type="double" />
			<param name="l1_rMin" value= "0.3" type="double" />
			<param name="l1_rMax" value= "20" type="double" />
			<param name="l2_frameName" value= "rear_laser" type="str" />
			<param name="l2_nRays" value= "133" type="int" />
			<param name="l2_aperture" value= "3.31" type="double" />
			<param name="l2_rMin" value= "0.3" type="double" />
			<param name="l2_rMax" value= "20" type="double" />

			<!--remap topic names -->
			<remap from="/$(env ROBOT)/localization3d_node/platform_data"
				to="/$(env ROBOT)/segway_rmp200_node/status" />
			<remap from="/$(env ROBOT)/iri_localization3d/platform_odom"
				to="/$(env ROBOT)/segway_rmp200_odom/odom" />
			<remap from="/$(env ROBOT)/iri_localization3d/laser1/scan"
				to="/$(env ROBOT)/front_laser/scan" />
			<remap from="/$(env ROBOT)/iri_localization3d/laser2/scan"
				to="/$(env ROBOT)/rear_laser/scan" />
			<remap from="/$(env ROBOT)/iri_localization3d/laser3/scan"
				to="/$(env ROBOT)/vertical_laser/scan" />

		</node>
	</group>

	<!-- rviz configurated for debug localization -->
	<node name="rviz" 
		pkg="rviz" 
		type="rviz" 
		args="-d $(find iri_localization3d)/launch/debugLoc.vcg">
	</node>

</launch>

<!-- args="-d $(find iri_localization3d)/launch/debugLoc.vcg" -->
<!-- args="-d $(find tibi_dabo_operations)/launch/nav_sensors.vcg"> -->
